carloskvasir/kaggle_auto_login
# Feature/gcp terraform

# 🚀 GCP Infrastructure for Automated Kaggle Login

## 🔍 Changes
- Add complete GCP infrastructure using Terraform
- Configure Cloud Function for automated login
- Setup Cloud Scheduler for daily execution
- Implement secure credential management via Secret Manager
- Standardize environment variables with KAGGLE_ prefix
- Improve documentation and code organization

## 🛠️ Technical Details
- Cloud Function with Python runtime
- Daily scheduler trigger
- Secure credential storage in Secret Manager
- IAM with minimal required permissions
- HTTPS-only communication

## 📝 Testing Steps
1. Copy `.env.example` to `.env`
2. Add Kaggle credentials
3. Deploy using Terraform
4. Verify Cloud Function execution
5. Check Kaggle login status

## 🔒 Security
- Credentials stored securely in Secret Manager
- Minimal IAM permissions
- Environment variables standardization
- HTTPS enforcement

## 📚 Documentation
- Updated README with deployment steps
- Added architecture diagram
- Included troubleshooting guide
